api/apps/api/src/modules/projects/output-project-summaries/output-project-summaries.service.ts

```diff
@@ -0,0 +1,291 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Scenario } from '@marxan-api/modules/scenarios/scenario.api.entity';
+import { BestSolutionDataService } from '@marxan-api/modules/projects/output-project-summaries/solution-data/best-solution-data.service';
+import { SummedSolutionDataService } from '@marxan-api/modules/projects/output-project-summaries/solution-data/summed-solution-data.service';
+import { ProjectsPuEntity } from '@marxan-jobs/planning-unit-geometry';
+import { CsvFormatterStream, FormatterRow, write } from 'fast-csv';
+import { execSync } from 'child_process';
+import { createWriteStream, existsSync, readFileSync } from 'fs';
+import { pipeline } from 'stream/promises';
+import { join as joinPath } from 'path';
+import { Injectable, Logger } from '@nestjs/common';
+import { OutputProjectSummaryApiEntity } from '@marxan/output-project-summaries';
+import { mkdir, rm } from 'fs/promises';
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { AppConfig } from '@marxan-api/utils/config.utils';
+
+export const outputProjectSummaryFolder = 'output-project-summary/';
+export const outputProjectSummaryFilename = 'summary.csv';
+export const outputProjectSummaryScenariosFilename = 'scenarios.csv';
+
+@Injectable()
+export class OutputProjectSummariesService {
+  private readonly logger = new Logger(OutputProjectSummariesService.name);
+
+  constructor(
+    private readonly bestSolutionsDataService: BestSolutionDataService,
+    private readonly summedSolutionsDataService: SummedSolutionDataService,
+    @InjectRepository(Scenario)
+    private readonly scenarioRepo: Repository<Scenario>,
+    @InjectRepository(ProjectsPuEntity, DbConnections.geoprocessingDB)
+    private readonly projectsPU: Repository<ProjectsPuEntity>,
+    @InjectRepository(OutputProjectSummaryApiEntity)
+    private readonly outputSummaryRepo: Repository<OutputProjectSummaryApiEntity>,
+  ) {}
+
+  async saveSummaryForProjectOfScenario(scenarioId: string): Promise<void> {
+    const { projectId } = await this.scenarioRepo.findOneOrFail({
+      where: { id: scenarioId },
+    });
+
+    const planningUnits = (await this.projectsPU.find({ where: { projectId } }))
+      .map((pu) => pu.puid)
+      .sort();
+
+    const scenarios = await this.scenarioRepo.find({
+      select: { id: true, projectScenarioId: true, name: true },
+      where: { projectId },
+    });
+
+    const bestSolutionData = await this.getBestSolutionDataForScenarios(
+      scenarios,
+    );
+    const summedSolutionData = await this.getSummedSolutionDataForScenarios(
+      scenarios,
+    );
+
+    const workingDirectory = OutputProjectSummariesService.getProjectTempFolder(
+      projectId,
+    );
```

Comment:
I think we discussed that this may be highly unlikely, but... wouldn't it be safer to use a unique directory each time this code is run?

Reply:
if so, I would still do it in a separate "refinement" PR: for the moment I'd say this is ok, in the interest of moving the PR on, in any case.

Reply:
also, it's not just a case of "if several of these project summaries are compiled more or less at the same time for a single project", but also (tin foil hat on), if anywhere else in the code someone adds _something_ later on that uses the same path structure for something completely unrelated, then boom.

tl;dr there are very good reasons for `mktemp` to be an essential tool ;)

Reply:
Hmm, I had the temp folders to be based on the projectId so that it could be more easily identified in case of debugging and looking at the files in staging/production, but you got a good point there.
I'll do it on a separate PR then.